### ISSUE TITLE
Display bug using vim with exec

### ISSUE THREAD

**Issue by rvrignaud** [kubernetes/kubernetes#14711]
Hello,

I'm using GKE (1.0.6) and I have a weird display bug using vim in exec -ti.
An image worth a thousand words, the gif was done using a "fluentd-cloud-logging" pod based on ubuntu-14.04.
![5dd5b3485226687d52fe0de611996510](https://cloud.githubusercontent.com/assets/525763/10156095/8f5e1a28-667d-11e5-801f-2243a6cdf22f.gif)

It may be os related as the same test with with the "monitoring-heapster-v6" pod doesn't show the problem.
I have the same problem with my own pods using ubuntu.
Note that connecting to the node and using docker exec -ti doesn't show the problem

**Comment by bnprss** [comment]
export TERM=linux
 before anything should fix vim (ans others).

**Comment by rvrignaud** [comment]
Thank you for the workaround. This seems to works well. Any idea of why this is behaving not in the same way between docker exec and kubectl exec ?

**Comment by bnprss** [comment]
Hum... Docker should set this env variable for you? It's the only possibility I see.

**Comment by mikedanese** [comment]
cc @ncdc @smarterclayton

**Comment by smarterclayton** [comment]
You may have to specify tty on the container to get that - I vaguely recall that being a docker "feature".  Set container .tty and see if that adds that variable

**Comment by rvrignaud** [comment]
I'm sorry I didn't get what you mean by "Set container .tty and see if that adds that variable"

**Comment by smarterclayton** [comment]
Really just, if you docker run -t does that add TERM automatically.  The
container tty field is on the pod, but it's just resulting in -t for Docker.

On Sep 29, 2015, at 11:04 AM, Romain Vrignaud <notifications@github.com>
wrote:

I'm sorry I didn't get what you mean by "Set container .tty and see if that
adds that variable"

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/14711#issuecomment-143996388>
.

**Comment by ncdc** [comment]
I think this is because the exec code in Kubernetes doesn't currently handle `SIGWINCH`. I did a POC to implement this and I'm pretty sure it fixed the display issue reported here. In my prototype, I created a new stream to send the window width & height over. The code is here: https://github.com/ncdc/origin/commit/13220dbb5208df5659568e282bc8483079463d2c. I do plan on bringing this into Kubernetes at some point, but I want to discuss the design first.
